### PR TITLE
CY-1598 Update the single manager status command

### PR DESCRIPTION
**Pull request by inbala**
* Check the connection of RabbitMQ and not only the
  service

* Don't check the DB connection because passing our
  authentication implies PostgreSQL is healthy

* Check syncthing functionallity - valid when at least
  one device was seen recently

* Change the response format

### CODE REVIEW

File: rest-service/manager_rest/rest/resources_v3_1/status.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            current_app.logger.error(
+                'Syncthing check failed with {err_type}: {err_msg}'.format(
+                    err_type=type(err),
+                    err_msg=str(err),
+                )
+            )
+            self._add_or_update_service(services,
+                                        'File Sync Service',
+                                        INACTIVE_STATE)
+            return FAIL_STATE
```
Comment (tyacbovi) [inline rest-service/manager_rest/rest/resources_v3_1/status.py]: This should be inactive ? No?

File: rest-service/manager_rest/rest/resources_v3_1/status.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+            # Syncthing is valid when at least one device was seen recently
+            if last_seen > min_last_seen:
+                syncthing_status = ACTIVE_STATE
+                break
+
+        self._add_or_update_service(services,
+                                    'File Sync Service',
+                                    syncthing_status)
+        return self._get_updated_status(manager_status, syncthing_status)
```
Comment (tyacbovi) [inline rest-service/manager_rest/rest/resources_v3_1/status.py]: Like I said this should not be inside

File: rest-service/manager_rest/test/endpoints/test_status.py [hunk 3]
```diff
@@ -21,14 +21,16 @@
 from manager_rest.storage import management_models
 
 
-@attr(client_min_version=1, client_max_version=base_test.LATEST_API_VERSION)
+@attr(client_min_version=3.1, client_max_version=base_test.LATEST_API_VERSION)
```
Comment (tyacbovi) [inline rest-service/manager_rest/test/endpoints/test_status.py]: Should keep old tests

File: rest-service/manager_rest/test/endpoints/test_status.py [hunk 4]
```diff
@@ -13,22 +13,40 @@
...
 
 from manager_rest.test import base_test
 from manager_rest.test.attribute import attr
 from manager_rest.storage import management_models
 
 
-@attr(client_min_version=1, client_max_version=base_test.LATEST_API_VERSION)
-class StatusTestCase(base_test.BaseServerTestCase):
+@attr(client_min_version=1, client_max_version=3)
+class PreviousStatusTestCase(base_test.BaseServerTestCase):
```
Comment (tyacbovi) [inline rest-service/manager_rest/test/endpoints/test_status.py]: PreviousStatusTestCase => StatusV1TestCase

File: rest-service/manager_rest/test/endpoints/test_status.py [hunk 5]
```diff
@@ -13,22 +13,40 @@
...
 
     def test_get_status(self):
-        self._assert_last_login_time_value(None)
+        before_status = datetime.utcnow()
         result = self.client.manager.get_status()
+        last_login_at = self.sm.get(management_models.User,
+                                    current_user.id).last_login_at
+        after_status = datetime.strptime(last_login_at,
```
Comment (tyacbovi) [inline rest-service/manager_rest/test/endpoints/test_status.py]: How did it work before?

File: rest-service/manager_rest/rest/resources_v3_1/status.py [hunk 6]
```diff
@@ -147,8 +147,8 @@ def _check_syncthing(self, services):
             self._add_or_update_service(services, display_name, status)
             return status
 
-        interval = syncthing_config['options']['reconnectionIntervalS']
-        min_last_seen = datetime.utcnow() - timedelta(seconds=interval + 1)
+        interval = syncthing_config['options']['reconnectionIntervalS'] + 1
```
Comment (tyacbovi) [inline rest-service/manager_rest/rest/resources_v3_1/status.py]: Mabe put a comment of the reason here

File: rest-service/manager_rest/rest/resources_v1/executions.py [hunk 7]
```diff
@@ -121,6 +123,16 @@ def post(self, **kwargs):
             scheduled_time=scheduled_time)
         return execution, 201
 
+    def _parse_scheduled_time(self, scheduled_time):
```
Comment (geokala) [inline rest-service/manager_rest/rest/resources_v1/executions.py]: Shouldn't this be in a separate PR?

File: rest-service/manager_rest/rest/resources_v1/status.py [hunk 8]
```diff
@@ -78,32 +78,34 @@ def get(self, **kwargs):
...
+    # If we reach here then the service is optional and not installed
+    return False
 
-    def _get_systemd_manager_services(self):
-        """Services the status of which we keep track of.
 
-        :return: a dict of {service_name: label}
-        """
-        services = {}
+def get_systemd_manager_services(base_services=BASE_SERVICES,
```
Comment (geokala) [inline rest-service/manager_rest/rest/resources_v1/status.py]: We shouldn't be setting default arg values to mutable values. At the moment, we don't break it, but it's generally bad practise as it can lead to unexpected interesting situations (TM) in future.

File: rest-service/manager_rest/rest/resources_v3_1/status.py [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+
+        self._add_or_update_service(services, display_name, status)
+        return status
+
+    def _get_manager_status(self, systemd_statuses, rabbitmq_status,
+                            syncthing_status):
+        statuses = systemd_statuses.union({rabbitmq_status, syncthing_status})
+
+        # statuses set can maximum have 2 statuses Active and Inactive
+        return HEALTHY_STATE if len(statuses) == 1 and \
```
Comment (geokala) [inline rest-service/manager_rest/rest/resources_v3_1/status.py]: Maybe better as: return HEALTHY_STATE if all(status == HEALTHY_STATE for status in statuses) else FAIL_STATE
?

File: rest-service/manager_rest/rest/resources_v1/executions.py (hunk 7 above)
Comment (inbala) [reply rest-service/manager_rest/rest/resources_v1/executions.py]: I needed it for the syncthing check in GET /status

File: rest-service/manager_rest/rest/resources_v3_1/status.py (hunk 9 above)
Comment (inbala) [reply rest-service/manager_rest/rest/resources_v3_1/status.py]: But then I don't use the advantage of the set

Comment (geokala) [reply rest-service/manager_rest/rest/resources_v3_1/status.py]: I think it's more readable with the all(x) approach though? Is there somewhere else that we get significant advantage from making this a set?

File: rest-service/manager_rest/rest/resources_v1/executions.py (hunk 7 above)
Comment (geokala) [reply rest-service/manager_rest/rest/resources_v1/executions.py]: Fair enough, I'll mark this resolved then.
